### PR TITLE
Fix addAssociation, addDiscriminatorColumn and addOverride on DoctrineCollector

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because without this fix it is imposible to use this feature.

I am trying to use this bundle to generate dynamic mapping here: https://github.com/Runroom/RunroomSamplesBundle/blob/master/src/DependencyInjection/RunroomSamplesExtension.php#L43

I had to stop using it and return to the EasyExtendsBundle way. Why? Because I keep getting an error similar to this: https://stackoverflow.com/questions/44323315/symfony-compiler-pass-i-want-to-add-method-call-to-logger-service 

I think the problem is trying to use the compiler pass to add calls with objects without using definitions: https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php#L41

That `$options` without this patch is an array containing an object, for some reason, Symfony does not like it.

I do not fully understand the problem, but I found this solution. I open the PR to discuss if it is the best approach or not. I know it is a BC break, but we should consider that actually this feature is imposible to use without an error. Of course, if the fix is accepted and test need fixing, I can do it.

## Changelog

```markdown
### Fixed
- Fix modifying entities (associations, discriminator columns and overrides) with DoctrineCollector
```

### CODE REVIEW

**Comment by core23** [comment]
Can you please add a test to cover this bug

**Comment by jordisala1991** [comment]
> Can you please add a test to cover this bug

I just added a test that fails without any code changes.

**Comment by jordisala1991** [comment]
Then I fix this test, but it is not enough. The problem is still there, but not visible on this test.

**Comment by jordisala1991** [comment]
This one is the real problem @core23

**Comment by jordisala1991** [comment]
Sorry, but it was hard to explain with just one commit. There are 2 problems. One is a discrepancy between the DoctrineORMMapper method `addAssociation`, because it is expecting and OptionsBuilder but on the CompilerPass it receives an array.

But even fixing this bug, there is the other issue where the dependency injection do not like to inject an object directly.

**Comment by franmomu** [comment]
Interesting! Apparently the calls added by the compiler pass, shouldn't have an object as a parameter, I guess that when Symfony dumps the container doesn't know how to manage those objects which is not a Reference, so I guess the same happens to `addDiscriminatorColumn` 🤔

**Comment by franmomu** [comment]
I guess one way to solve it is to [modify DoctrineORMMapper](https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/ORM/DoctrineORMMapper.php#L69) to remove the `OptionsBuilder` type and [ColumnDefinitionBuilder](https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/ORM/DoctrineORMMapper.php#L95) type and call them with an array.

**Comment by jordisala1991** [comment]
> I guess one way to solve it is to [modify DoctrineORMMapper](https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/ORM/DoctrineORMMapper.php#L69) to remove the `OptionsBuilder` type and [ColumnDefinitionBuilder](https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/ORM/DoctrineORMMapper.php#L95) type and call them with an array.

Yes, That what I thought. Initially this one is the fix for addAssociation only, but I can extend it to ther others. I just want to make sure we are doing the right changes, because there are also a miss on a variable name

File: src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php [hunk 1]
```diff
@@ -37,8 +37,8 @@ public function process(ContainerBuilder $container): void
         $collector = DoctrineCollector::getInstance();
 
         foreach ($collector->getAssociations() as $class => $associations) {
-            foreach ($associations as $field => $options) {
-                $mapper->addMethodCall('addAssociation', [$class, $field, $options]);
+            foreach ($associations as $type => $options) {
```
Comment (jordisala1991) [inline src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php]: this only is a type, not a field name. (It contains: 'mapManyToOne')

**Comment by jordisala1991** [comment]
This one should be fine for addAssociation. Do you want me to fix the others here or on other PRs?

**Comment by franmomu** [comment]
> This one should be fine for addAssociation. Do you want me to fix the others here or on other PRs?

Nice! I would fix them here as well.

**Comment by jordisala1991** [comment]
Done! I think it is a great increase on coverage! Should we note the BC break (unavoidable I think) somewhere?

File: tests/Mapper/ORM/DoctrineORMMapperTest.php [hunk 2]
```diff
@@ -14,17 +14,41 @@
...
      * @var ClassMetadataInfo
      */
     private $metadata;
 
+    public static function setUpBeforeClass(): void
+    {
+        $options = OptionsBuilder::createManyToOne('relation', TestRelatedEntity::class)
+            ->add('joinColumns', [['referencedColumnName' => 'id']]);
+
+        $override = OptionsBuilder::create()
```
Comment (jordisala1991) [inline tests/Mapper/ORM/DoctrineORMMapperTest.php]: Side note: we deprecrated the direct create here: https://github.com/sonata-project/sonata-doctrine-extensions/pull/177
But for adding overrides, you need to use the options builder and it might be an override of a non relation field. So the methods currently available are not suitable for this use case. Maybe we need to add one more method or undeprecate this create.

Comment (greg0ire) [reply tests/Mapper/ORM/DoctrineORMMapperTest.php]: cc @franmomu

Comment (franmomu) [reply tests/Mapper/ORM/DoctrineORMMapperTest.php]: Maybe we could create a class for override, I've never used them, which are the options? And maybe since there has been no release we could create AssociationBuilder which extends OptionsBuilder (to be BC) and has all the methods I added.

Comment (jordisala1991) [reply tests/Mapper/ORM/DoctrineORMMapperTest.php]: IMO that should belong to another PR. Is this one ready!?

File: tests/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPassTest.php [hunk 3]
```diff
@@ -61,4 +66,32 @@ public function testDefinitionsNotRemoved()
 
         $this->assertContainerBuilderHasService('sonata.doctrine.mapper');
     }
+
+    public function testAssociationMapping()
```
Comment (franmomu) [inline tests/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPassTest.php]:
```suggestion
    public function testAssociationMapping(): void
```

File: tests/Mapper/ORM/DoctrineORMMapperTest.php (hunk 2 above)
Comment (franmomu) [reply tests/Mapper/ORM/DoctrineORMMapperTest.php]: Sure it should belong to another PR, just asking about docs to create that class.

File: tests/Fixtures/App/Kernel.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    public function getProjectDir(): string
+    {
+        return __DIR__;
+    }
+
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    {
+        $configDir = $this->getProjectDir() . '/config';
+
+        $container->addResource(new FileResource($configDir . '/bundles.php'));
```
Comment (franmomu) [inline tests/Fixtures/App/Kernel.php]: out of curiosity, what's this line for?

Comment (jordisala1991) [reply tests/Fixtures/App/Kernel.php]: This line is needed to load the bundles php file in order to test the bundle on a “real” app env

File: tests/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPassTest.php (hunk 3 above)
Comment (jordisala1991) [reply tests/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPassTest.php]: Fixed

File: tests/Mapper/ORM/DoctrineORMMapperTest.php [hunk 5]
```diff
@@ -80,4 +104,13 @@ public function testInheritanceTypes(): void
         $m->setAccessible(true);
         $m->invoke($mapper, $this->metadata);
     }
+
+    public function testDoctrineMappingLoaded(): void
```
Comment (franmomu) [inline tests/Mapper/ORM/DoctrineORMMapperTest.php]: IMO this test should belong to a separated class inside Symfony as we are mixing here unit and functional tests.

Comment (jordisala1991) [reply tests/Mapper/ORM/DoctrineORMMapperTest.php]: IMO this class does not have useful tests. What about removing the other tests and keep only this one? Adding the 2 missing test: discriminators and inheritanceType should cover the same as before but with better coverage and testing that actually this code works.

Comment (franmomu) [reply tests/Mapper/ORM/DoctrineORMMapperTest.php]: That's another way of seeing it 🤣I didn't pay attention to the other tests, you're right 👍

File: src/Mapper/DoctrineCollector.php [hunk 6]
```diff
@@ -95,11 +95,11 @@ public function addDiscriminator(string $class, string $key, string $discriminat
     public function addDiscriminatorColumn(string $class, ColumnDefinitionBuilder $columnDef): void
     {
         if (!isset($this->discriminatorColumns[$class])) {
-            $this->discriminatorColumns[$class] = $columnDef;
+            $this->discriminatorColumns[$class] = $columnDef->getOptions();
         }
     }
 
-    public function addInheritanceType(string $class, string $type): void
+    public function addInheritanceType(string $class, int $type): void
```
Comment (jordisala1991) [inline src/Mapper/DoctrineCollector.php]: Had to add another BC breack, this type should be int, otherwise it does not match with the DoctrineORMMapper (https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/ORM/DoctrineORMMapper.php#L105) or the   ClassMetadataInfo (https://github.com/doctrine/orm/blob/v2.7.2/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php#L69)

**Comment by jordisala1991** [comment]
I think I fixed all the bugs on the DoctrineORMMapper now.

File: src/Mapper/DoctrineCollector.php (hunk 6 above)
Comment (core23) [reply src/Mapper/DoctrineCollector.php]: Can you find a BC way for this?

File: src/Mapper/ORM/DoctrineORMMapper.php [hunk 7]
```diff
@@ -66,13 +64,13 @@ public function getSubscribedEvents(): array
         ];
     }
 
-    public function addAssociation(string $class, string $field, OptionsBuilder $options): void
+    public function addAssociation(string $class, string $type, array $options): void
```
Comment (core23) [inline src/Mapper/ORM/DoctrineORMMapper.php]: Can you find a BC way for this?

Comment (jordisala1991) [reply src/Mapper/ORM/DoctrineORMMapper.php]: I think the only way is to remove the previous typing, but not adding the correcto one. I thought that since the bug is caused by this incorrect type we could change it without BC, because the previous type couldn't work on a real app. wdyt?

**Comment by jordisala1991** [comment]
I removed the typehint to avoid a BC Break.

File: src/Mapper/DoctrineCollector.php [hunk 8]
```diff
@@ -95,12 +95,22 @@ public function addDiscriminator(string $class, string $key, string $discriminat
...
-    public function addInheritanceType(string $class, string $type): void
+    /**
+     * @param int $type
+     */
+    public function addInheritanceType(string $class, $type): void
     {
+        // NEXT_MAJOR: Move int check to method signature
+        if (!\is_int($type)) {
+            throw new \InvalidArgumentException(
+                'Argument 2 is not an integer, expecting an integer'
```
Comment (phansys) [inline src/Mapper/DoctrineCollector.php]: The argument declaration will produce an exception of type `\TypeError`, with a message similar to:
```suggestion
            throw new \TypeError(
                \sprintf('Argument 2 passed to %s() must be of the type int, %s given.', __METHOD__, is_object($type) ? get_class($type) : gettype($type));
```
The same applies for other exceptions added in this PR.

File: phpunit.xml.dist [hunk 9]
```diff
@@ -33,6 +33,7 @@ It's auto-generated by sonata-project/dev-kit package.
         <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[self]=0" />
         <env name="SYMFONY_PHPUNIT_REMOVE" value="symfony/yaml" />
         <env name="SYMFONY_PHPUNIT_VERSION" value="8.5.0" />
+        <env name="KERNEL_CLASS" value="Sonata\Doctrine\Tests\Fixtures\App\Kernel" />
```
Comment (phansys) [inline phpunit.xml.dist]: Maybe we could keep this list in alphabetical order.

Comment (VincentLanglet) [reply phpunit.xml.dist]: This file should not be edited as mention in top of it. It's coming from https://github.com/sonata-project/dev-kit/blob/master/project/phpunit.xml.dist

Comment (jordisala1991) [reply phpunit.xml.dist]: And how do I add this env var? Is there another way? Because symfony depends on this variable to load the kernel

File: src/Mapper/DoctrineCollector.php (hunk 8 above)
Comment (jordisala1991) [reply src/Mapper/DoctrineCollector.php]: Fixed

File: phpunit.xml.dist (hunk 9 above)
Comment (jordisala1991) [reply phpunit.xml.dist]: Solved with a workaround to not edit this file

File: tests/Fixtures/App/Kernel.php [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Doctrine\Tests\Fixtures\App;
```
Comment (phansys) [inline tests/Fixtures/App/Kernel.php]: Maybe we could respect the same directory / namespace structure than other bundles, where the `App\` NS is directly under `Tests\` ([ref](https://github.com/sonata-project/SonataAdminBundle/blob/604aaccf2a5b3fff4746f12d803bdd6c4fe9e76e/tests/App/AppKernel.php#L14)).

File: src/Mapper/ORM/DoctrineORMMapper.php [hunk 11]
```diff
@@ -66,13 +64,25 @@ public function getSubscribedEvents(): array
...
 
-    public function addAssociation(string $class, string $field, OptionsBuilder $options): void
+    /**
+     * @param array $options
+     */
+    public function addAssociation(string $class, string $type, $options): void
     {
+        // NEXT_MAJOR: Move array check to method signature
+        if (!\is_array($options)) {
+            throw new \TypeError(sprintf(
```
Comment (core23) [inline src/Mapper/ORM/DoctrineORMMapper.php]: This is a BC break :( 

You could add a deprecation error to keep BC

File: src/Mapper/DoctrineCollector.php [hunk 12]
```diff
@@ -95,12 +95,24 @@ public function addDiscriminator(string $class, string $key, string $discriminat
...
-    public function addInheritanceType(string $class, string $type): void
-    {
+    /**
+     * @param int $type
+     */
+    public function addInheritanceType(string $class, $type): void
+    {
+        // NEXT_MAJOR: Move int check to method signature
+        if (!\is_int($type)) {
+            throw new \TypeError(sprintf(
```
Comment (VincentLanglet) [inline src/Mapper/DoctrineCollector.php]: This is a BC-break

File: src/Mapper/ORM/DoctrineORMMapper.php [hunk 13]
```diff
@@ -92,10 +102,22 @@ public function addDiscriminator(string $class, string $key, string $discriminat
...
 
-    public function addDiscriminatorColumn(string $class, ColumnDefinitionBuilder $columnDef): void
+    /**
+     * @param array $columnDef
+     */
+    public function addDiscriminatorColumn(string $class, $columnDef): void
     {
+        // NEXT_MAJOR: Move array check to method signature
+        if (!\is_array($columnDef)) {
+            throw new \TypeError(sprintf(
```
Comment (VincentLanglet) [inline src/Mapper/ORM/DoctrineORMMapper.php]: This is a BC break

File: src/Mapper/ORM/DoctrineORMMapper.php [hunk 14]
```diff
@@ -145,13 +167,25 @@ public function addUnique(string $class, string $name, array $columns): void
...
 
-    public function addOverride(string $class, string $type, OptionsBuilder $options): void
+    /**
+     * @param array $options
+     */
+    public function addOverride(string $class, string $type, $options): void
     {
+        // NEXT_MAJOR: Move array check to method signature
+        if (!\is_array($options)) {
+            throw new \TypeError(sprintf(
```
Comment (VincentLanglet) [inline src/Mapper/ORM/DoctrineORMMapper.php]: This is a BC break

File: src/Mapper/ORM/DoctrineORMMapper.php (hunk 11 above)
Comment (phansys) [reply src/Mapper/ORM/DoctrineORMMapper.php]: I agree, but I also think a new exception like this could be considered as BC compliant if the previous code throws a similar exception sooner or later because the unsupported type, but I don't know if it is the case.

Comment (VincentLanglet) [reply src/Mapper/ORM/DoctrineORMMapper.php]: This type used was changed.
From
```
public function addAssociation(string $class, string $field, OptionsBuilder $options): void
```
To
```
public function addAssociation(string $class, string $field, array $options): void
```

Comment (phansys) [reply src/Mapper/ORM/DoctrineORMMapper.php]: You're right. Given the signature change I was assuming the `OptionsBuilder` declaration was added in a yet unreleased version, but I see that's not the case:

https://github.com/sonata-project/sonata-doctrine-extensions/blob/9f39109af81561f8175a3bcece454724a8d9e053/src/Mapper/ORM/DoctrineORMMapper.php#L69

This is really a BC break.
Thanks for pointing.

Comment (jordisala1991) [reply src/Mapper/ORM/DoctrineORMMapper.php]: Why is this a BC break? this Pull request is about fixing a bug because of this type hint.

**Comment by jordisala1991** [comment]
Please explain me how to proceed. This pull request is about fixing a feature that currently is not working and cannot be used on any app (so no one is able to use this code -> this change cannot break any app using this code, because no one can use this code).

File: src/Mapper/ORM/DoctrineORMMapper.php (hunk 11 above)
Comment (VincentLanglet) [reply src/Mapper/ORM/DoctrineORMMapper.php]: The function alone could still be used by someone with an OptionBuilder as a third argument.

You just have to trigger a deprecation like this https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AbstractAdmin.php#L1079 in order to be BC.

**Comment by VincentLanglet** [comment]
> Please explain me how to proceed. This pull request is about fixing a feature that currently is not working and cannot be used on any app (so no one is able to use this code -> this change cannot break any app using this code, because no one can use this code).

You just have to trigger deprecation instead of returning a TypeError

**Comment by jordisala1991** [comment]
> You just have to trigger deprecation instead of returning a TypeError

But what is the point, the code is unusable

**Comment by phansys** [comment]
The change from `array` to `OptionsBuilder` (which seems a BC break) was made [here](https://github.com/sonata-project/sonata-doctrine-extensions/commit/2e1c770eb9400221228faad6b3d9027b1f9b3b11#diff-a30dc7aeee437b788323d235da99723dR158).
Maybe @core23 could share some context.

**Comment by core23** [comment]
> The change from `array` to `OptionsBuilder` (which seems a BC break) was made [here](https://github.com/sonata-project/sonata-doctrine-extensions/commit/2e1c770eb9400221228faad6b3d9027b1f9b3b11#diff-a30dc7aeee437b788323d235da99723dR158).
> Maybe @core23 could share some context.

That's right. The mentioned PR introduced a BC break which should be reverted. To stay BC we should support the `OptionsBuilder` and `array`. The `array` argument should throw a deprecation notice.

**Comment by jordisala1991** [comment]
All changes applied, ready to review

**Comment by core23** [comment]
I would prefer this one: https://github.com/sonata-project/sonata-doctrine-extensions/pull/185

The API is more clear in the other PR.

**Comment by jordisala1991** [comment]
My Pr is not about api change but a bugfix

**Comment by jordisala1991** [comment]
The other PR is not fixing the bug and it does not provide the needed functional tests I am adding here

**Comment by jordisala1991** [comment]
@phansys @core23 can you review please?

File: src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php [hunk 15]
```diff
@@ -37,8 +37,8 @@ public function process(ContainerBuilder $container): void
...
+                $mapper->addMethodCall('addAssociation', [$class, $type, $options]);
```
Comment (OskarStark) [inline src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php:41]: I would do this variable renaming in a separate PR

File: tests/Mapper/DoctrineCollectorTest.php [hunk 16]
```diff
@@ -46,7 +47,7 @@ public function testClear(): void
         $collector = DoctrineCollector::getInstance();
         $collector->addIndex(stdClass::class, 'name', ['column']);
         $collector->addUnique(stdClass::class, 'name', ['column']);
-        $collector->addInheritanceType(stdClass::class, 'type');
+        $collector->addInheritanceType(stdClass::class, ClassMetadataInfo::INHERITANCE_TYPE_SINGLE_TABLE);
```
Comment (OskarStark) [inline tests/Mapper/DoctrineCollectorTest.php:50]: I would do this change in a separate PR

File: src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php (hunk 15 above)
Comment (jordisala1991) [reply src/Bridge/Symfony/DependencyInjection/Compiler/MapperCompilerPass.php]: Done

File: tests/Mapper/DoctrineCollectorTest.php (hunk 16 above)
Comment (jordisala1991) [reply tests/Mapper/DoctrineCollectorTest.php]: Done

Comment (jordisala1991) [reply tests/Mapper/DoctrineCollectorTest.php]: This change is necessary because:

1x: Passing other type than int as argument 2 for method Sonata\Doctrine\Mapper\DoctrineCollector::addInheritanceType() is deprecated since sonata-project/doctrine-extensions 1.x. It will accept only int in version 2.0.

**Comment by core23** [comment]
> I do not fully understand the problem, but I found this solution. I open the PR to discuss if it is the best approach or not. I know it is a BC break, but we should consider that actually this feature is imposible to use without an error. Of course, if the fix is accepted and test need fixing, I can do it.

Maybe I got this wrong, but it looks like the problem is not the actual `DoctrineCollector`, but the symfony compiler pass. IMHO you should fix the compiler pass instead of making the API lenient. 

Keep in mind, that this not a symfony bundle itself. This is a library that has a symfony bridge.

**Comment by jordisala1991** [comment]
The only thing that can be pass on a `addMethodCall` is either a Reference class or a basic type. Passing a class is not allowed. And to pass a Reference you should be doing something like build the object right there with all the calls again, which I don't think is possible.

I am open to listen for differents solutions, but in the meantime, this bundle is not usable.

**Comment by core23** [comment]
> The only thing that can be pass on a `addMethodCall` is either a Reference class or a basic type. Passing a class is not allowed. And to pass a Reference you should be doing something like build the object right there with all the calls again, which I don't think is possible.
> 
> https://github.com/symfony/dependency-injection/blob/master/Dumper/XmlDumper.php#L384

Why is this an `XmlDumper` problem? The compiler pass uses this class https://github.com/symfony/dependency-injection/blob/master/Definition.php#L354

**Comment by jordisala1991** [comment]
It is an XmlDumper problem because it is happening when the container tries to compile itself to xml (on the cache folder). This is why a functional test booting a kernel is needed to catch the bug.

File: UPGRADE-1.x.md [hunk 17]
```diff
@@ -1,13 +1,28 @@
 UPGRADE 1.x
 ===========
 
+### `Sonata\Doctrine\Mapper\DoctrineCollector`
+
+Not passing an integer as the second parameter of
+`addInheritanceType` is deprecated, you should pass
+an integer.
```
Comment (phansys) [inline UPGRADE-1.x.md:8]:
```suggestion
Deprecated passing other type than integer as argument 2 for `addInheritanceType()`.
```

File: UPGRADE-1.x.md [hunk 18]
```diff
@@ -1,13 +1,28 @@
...
+
+### `Sonata\Doctrine\Mapper\ORM\DoctrineORMMapper`
+
+Not passing an array as the third parameter of
+`addAssociation` is deprecated, you should pass an array.
+Not passing an array as the second parameter of
+`addDiscriminatorColumn` is deprecated, yo should pass an array.
+Not passing an array as the third parameter of
+`addOverride` is deprecated, you should pass an array.
```
Comment (phansys) [inline UPGRADE-1.x.md:17]:
```suggestion
Deprecated passing other type than array as argument 3 for `addAssociation()`.
Deprecated passing other type than array as argument 2 for `addDiscriminatorColumn()`.
Deprecated passing other type than array as argument 3 for `addOverride()`.
```

**Comment by jordisala1991** [comment]
Ping @core23 @phansys , can you review?

Errors before this PR:
https://travis-ci.org/github/sonata-project/SonataMediaBundle/jobs/704863805#L804
https://travis-ci.org/github/sonata-project/SonataPageBundle/jobs/704863422#L738

**Comment by jordisala1991** [comment]
Ping @sonata-project/contributors , can we unblock this?

**Comment by jordisala1991** [comment]
@core23 your review is needed here, it cannot be merged otherwise

**Comment by jordisala1991** [comment]
ping again... @sonata-project/contributors can we unblock this?

**Comment by VincentLanglet** [comment]
> ping again... @sonata-project/contributors can we unblock this?

You need the @core23 approval or someone with enough admin rights like @greg0ire to merge this PR.

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !
